tracing-core/src/dispatch.rs

```diff
@@ -853,6 +857,10 @@ impl Collect for NoCollector {
 
     fn enter(&self, _span: &span::Id) {}
     fn exit(&self, _span: &span::Id) {}
+
+    fn current_span(&self) -> span::Current {
+        span::Current::unknown()
```

Comment:
nit, this doesn't really matter: i think it _might_ be more correct for the no-op collector to return `none` rather than `unknown` — the intended semantics is that all spans are _disabled_ when there's no collector, so we are never in a span. with that said, it doesn't really make a difference in practice.